jmeter/src/main/java/org/apache/jmeter/protocol/http/proxy/JMeterProxy.java

```diff
@@ -342,17 +336,14 @@ public void run()
      *
      * @param _clientSocket  the socket connection to the client
      * @param _target        the ProxyControl which will receive the generated sampler
-     * @param _pageEncodings reference to the Map of Deamon, with mappings from page urls to encoding used
-     * @param _formEncodings reference to the Map of Deamon, with mappings from form action urls to encoding used
      */
-    public void configure(Socket _clientSocket, JMeterProxyControl _target, Map<String, String> _pageEncodings,
-            Map<String, String> _formEncodings)
+    public void configure(Socket _clientSocket, JMeterProxyControl _target)
     {
         this.target = _target;
         this.clientSocket = _clientSocket;
         this.captureHttpHeaders = _target.getCaptureHttpHeaders();
-        this.pageEncodings = _pageEncodings;
-        this.formEncodings = _formEncodings;
+        this.pageEncodings = Collections.synchronizedMap(new HashMap<String, String>());
+        this.formEncodings = Collections.synchronizedMap(new HashMap<String, String>());
```

Comment:
I'm don't think, that we should touch any code, that was copypasted from JMeter sources. I do this modification, because there was a TODO at JMeterDaemon.java:110, and, in my vision, it was to reduce the memory consumption